snooty/rstparser.py

```diff
@@ -485,6 +486,58 @@ def run(self) -> List[docutils.nodes.Node]:
         return [node]
 
 
+class TocTreeDirective(docutils.parsers.rst.Directive):
+    """Special handling for toctree directives.
+
+    Produces a node that includes an `entries` property, represented as a list of objects. Each entry in entries includes:
+    - slug OR url: a string representing the absolute url or path of the page to navigate to
+    - [optional] title: a string representing the title to use in the TOC sidebar
+    """
+
+    required_arguments = 0
+    optional_arguments = 0
+    has_content = True
+    final_argument_whitespace = True
+    option_spec = {
+        "caption": str,
+        "hidden": util.option_flag,
+        "includehidden": util.option_flag,
+        "maxdepth": int,
+        "titlesonly": util.option_flag,
```

Comment:
These are the only options utilized in our docs repos, but there are a few more (defined [here](https://www.sphinx-doc.org/en/master/usage/restructuredtext/directives.html#table-of-contents)). Do we want to support them all?

Reply:
We have some discretion about what to support, but in general we should try to support everything!